### PR TITLE
Run automatic `node-gyp rebuild`

### PR DESCRIPTION
**Pull request by BendingBender**
As described by: https://github.com/npm/cli/blob/f9abee79abe541226a249f50bdeec41317dd5712/docs/content/configuring-npm/package-json.md#default-values

### CODE REVIEW

**Comment by legobeat** [comment]
Hello @BendingBender - thank you for your contribution and apologies for leaving this hanging. If you are still interested in this, would you be able to reduce this to the changes needed for this feature (removing or splitting out for example the .nvmrc and .gitignore changes)?

**Comment by BendingBender** [comment]
OK, I've removed all the unnecessary changes and rebased to current main.

**Review by boneskull — DISMISSED** [review]
@BendingBender If you're still listening, I apologize.

This is so old that the URL in the PR body no longer points to useful information, so I have no idea why we should merge this.

- [ ] needs to be rebased onto `main`
- [ ] `rimraf` can be removed; use `fs.rm`

File: packages/allow-scripts/src/index.js [hunk 1]
```diff
@@ -392,6 +392,22 @@ async function loadAllPackageConfigurations({ rootDir }) {
...
+      let bindingGypExists = false
+      try {
+        await fs.stat(path.join(filePath, 'binding.gyp'))
+        bindingGypExists = true
+      } catch (err) {}
+
+      if (bindingGypExists) {
+        lifeCycleScripts.unshift('install')
+        depScripts.install = 'node-gyp rebuild'
+      }
```
Comment (boneskull) [inline packages/allow-scripts/src/index.js]:
```suggestion
      try {
        await fs.stat(path.join(filePath, 'binding.gyp'))
        lifeCycleScripts.unshift('install')
        depScripts.install = 'node-gyp rebuild'
      } catch {}
```

**Review by naugtur — APPROVED** [review]
Sorry for hanging on to this for so long. I'll merge to a branch and address any leftover reviews internally.

**Comment by naugtur** [comment]
@BendingBender expect it in the next release

**Pull request merged**